### PR TITLE
 Handle some TODOs in tests 

### PR DESCRIPTION
**Pull request by vdemeester**
Use more gotestyourself for `env.Patch`, and `icmd.RunCommand`

Also remove some `FIXME` comments on vendor.conf as they don't really make sense.

:lion:

### CODE REVIEW

File: e2e/container/run_test.go [hunk 1]
```diff
@@ -17,8 +16,8 @@ const registryPrefix = "registry:5000"
 func TestRunAttachedFromRemoteImageAndRemove(t *testing.T) {
 	image := createRemoteImage(t)
 
-	result := icmd.RunCmd(shell(t,
-		"docker run --rm %s echo this is output", image))
+	result := icmd.RunCommand("docker", "run", "--rm", image,
+		"echo", "this", "is", "output")
```
Comment (thaJeztah) [inline e2e/container/run_test.go]: Should "this is output" be a single string here?

Comment (vdemeester) [reply e2e/container/run_test.go]: Both work :stuck_out_tongue_winking_eye:

File: cli/command/plugin/install.go [hunk 2]
```diff
@@ -124,9 +124,8 @@ func buildPullConfig(ctx context.Context, dockerCli command.Cli, opts pluginOpti
 		Disabled:              opts.disable,
 		AcceptAllPermissions:  opts.grantPerms,
 		AcceptPermissionsFunc: acceptPrivileges(dockerCli, opts.remote),
-		// TODO: Rename PrivilegeFunc, it has nothing to do with privileges
```
Comment (thaJeztah) [inline cli/command/plugin/install.go]: This no longer relevant? 😇

Comment (vdemeester) [reply cli/command/plugin/install.go]: It may be, but the comment should be on the API struct (thus in `moby/moby`)

Comment (cpuguy83) [reply cli/command/plugin/install.go]: Why? This is referring to how to get auth credentials?

Comment (silvin-lubecki) [reply cli/command/plugin/install.go]: So did you move this comment to moby/moby? 😄

Comment (cpuguy83) [reply cli/command/plugin/install.go]: 🤦‍♂️

**Pull request merged**